### ISSUE TITLE
Empty Qt help window (Ubuntu 20)

### ISSUE THREAD

**Issue by wthaem** [Maproom/qmapshack#185]
**Important Notice:**

**Do not delete any of the sections. Answer them. If you really**
**think they do not apply to you give a short statement. Bug**
**reports that are not properly filed might be closed without**
**further notice.**

**Choose a good title**

What is a good title?
The title will be listed in the changelog when the issue is fixed. Do
you think someone else would understand what's fixed or done from your
title? Does it really name the real topic of the issue?

[ x] Did you choose a good title?

**Describe the bug**

While the help on Ubuntu 18.04 and on Windows 10 runs as expected there are problems when using Ubuntu 20. Here, pressing F1 opens an empty Qt help window (no HTML page displayed in right part, no TOC, no index).

**What have you done to circle down the problem?**

Checked with Ubuntu 18.04 and Windows 10, no problems here. All OS have different versions of Qt assistant!

**To Reproduce**

1. Start Ubuntu 20 in a virtual VMWare machine
1. Start QMS
1. Press F1 and check help window

**Expected behavior**

Help window should have complete content: TOC, index, home page shown at start.

**Screenshots**

not necessary

**Attachments**

none

**Tracebacks**

compare logfile entries shown later

**Desktop (please complete the following information):**

 - OS: Ubuntu 20 with VMWare 
 - QMapShack latest commit from repo
 - Window Manager: not known (the one coming with the Ubuntu distribution)

**Additional context**

* Same issue with QMapTool
* QMS for Ubuntu 18 and 20 both compiled from sources on equal commit level using compilation rules described in Ubuntu Wiki page
* Help files are at correct location
* Help files can be opened with Qt assistant version 5.12.8 (Ubuntu 20) and show correct content.
* Qt assistant version 5.9.5 used with Ubuntu 18.
* QMS logfile entries relevant for this issue:

~~~
2020-05-13 15:06:11.894 [debug] search help at: "/usr/local/share/doc/HTML/QMSHelp.qhc"
2020-05-13 15:06:11.941 [debug] QUrl("qthelp://qms/doc/doc/html/DocMain.html")
2020-05-13 15:06:11.941 [warning] QTextBrowser: No document for qthelp://qms/doc/doc/html/DocMain.html
2020-05-13 15:06:11.942 [warning] QMainWindow::addDockWidget: invalid 'area' argument
~~~

(no idea, if the last line belong to this issue)

The path to QMSHelp.qhc is correct. The internal path to DocMain.html is correct (no difference to Ubuntu 18 run).

The QTextBrowser warning doesn't appear in an Ubuntu 18 logfile.

**Comment by kiozen** [comment]
Looks like a problem of the used QT library. But this has to be solved by those using Qt 5.12.

**Comment by wthaem** [comment]
Any idea or proposal what to do next?

**Comment by kiozen** [comment]
Well the usual:

1. Check if there has been an API change. (This shouldn't happen, but who knows)
2. Check if there is a bug report and some advise to mitigate the problem.

**Comment by wthaem** [comment]
After some testing even using an example with minimal C++ code for calling Qt help I found a surprising result: the Ubuntu QMSHelp.* files in "/usr/local/share/doc/HTML/" were broken. Replacing them as superuser by better ones solved the issue.

I don't have a clue how this can happen. I pulled the latest source code including the help files from the remote dev repo. In the local repo the QMSHelp.* files are ok (checked this with assistant). Then I followed the Ubuntu compile steps described in the Wiki including the "sudo make install" step. The help files at there final location (the broken ones) are read-only after installation. I never used a "sudo ..." command to change these files.

Positive result of tests:

Qt help in QMS works as expected with the following systems:

Windows 10 + Qt 5.13.2
Ubuntu 18.04 + Qt 5.9.5
Ubuntu 20 + Qt 5.12.8
Linux Mint 19.2 + Qt 5.9.5

With this I don't see a need to keep this issue open.
